apps/examples/araplayer/AraPlayer.cpp

```diff
@@ -0,0 +1,262 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "AraPlayer.h"
+#include "AudioHttpDownloader.h"
+#include "MediaPlayerWrapper.h"
+#include "AudioPDStream.h"
+#include "AudioHASStream.h"
+#include "AudioLocalStream.h"
+#include "AudioWorker.h"
+
+AraPlayer::AraPlayer() :
+	mCurrStream(nullptr)
+{
+}
+
+AraPlayer::~AraPlayer()
+{
+}
+
+void AraPlayer::initWorker()
+{
+	DEBUG_TRACE(LOG_INFO, "worker initialized");
+	return notifySync();
+}
+
+void AraPlayer::init()
+{
+	std::unique_lock<std::mutex> lock(mCmdMtx);
+	AudioWorker &aw = AudioWorker::getWorker();
+	Timer::getInstance();
+	initCurl();
+	aw.setCurl(mCurl);
+	aw.startWorker();
+	aw.enQueue(&AraPlayer::initWorker, shared_from_this());
+	mCv.wait(lock);
+}
+
+void AraPlayer::notifySync()
+{
+	std::unique_lock<std::mutex> lock(mCmdMtx);
+	mCv.notify_one();
+}
+
+void AraPlayer::initCurl()
+{
+	curl_global_init(CURL_GLOBAL_DEFAULT);
+	mCurl = (void *)curl_multi_init();
+	curl_multi_perform((CURLM *)mCurl, &running);
+}
+
+// TO-DO msg will be changed in MessagePort CB.
+AudioStream *AraPlayer::createStream(pid_t pid, ara_player_msg_t *msg)
```

Comment:
Logic is very strange.
Player doesn't know PD or HAS using curl or not.
so Player should call something common one. 
For example, 
```
switch(msg->streamType)
case STREAM_LOCAL:
   stream = new AudioLocalStream;
   break;
case STREAM_HTTP_PROGRESSIVE:
   stream = new AudioPDStream;
   break;
}
stream->initStream(); //local will disable curl but other will enable curl.
...

also curl can be created inside of each stream, and httpdownloader use it.
So when stream is initialized, it can send pointer of curl to httpdownloader.
Then we don't have to use setMainCurl or getMainCurl...